test/integration/appconfig_manifest_test.go

```diff
@@ -0,0 +1,192 @@
+// +build integration
+
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/test/integration"
+	"github.com/crossplane/oam-kubernetes-runtime/apis/core"
+	"github.com/crossplane/oam-kubernetes-runtime/apis/core/v1alpha2"
+	"github.com/crossplane/oam-kubernetes-runtime/pkg/controller/oam"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func TestAppConfigControllerWithManifests(t *testing.T) {
+	cases := []struct {
```

Comment:
This test suite seems to rely on test orders and each case is not self-contained (no cleanup). Both are something that I usually advocate against. Any reason for this pattern?

Reply:
Another thing is that my understanding of a table driven test methodology is that each test in the table is testing a  different scenario in the overall test dimension.  IIRC, test 1 and 2 here are not testing anything but K8s api-server.  While test 3 and 4 are basically two steps of a single test. Test 5, 6 are clean up. I  don't follow how this is a table. It looks more like a single test case decomposed.

Reply:
If I were to write a table-driven test, each test case would be a different suite of appConfig yamls.  I am not sure why it is not possible in ginkgo or any other test framework.

Reply:
Yeah I was mostly experimenting here. In the past I have done more steps in each single case so I can condense here 👍.

Reply:
I don't mind more steps in a single test but it seems that they are still combined to be just one case. This is just not the table-driven test pattern I envisioned.

Reply:
I definitely agree that it is not idiomatic to have this many granular steps :) updating now